assets/src/stories-editor/components/editor-carousel/index.js

```diff
@@ -24,74 +23,87 @@ import './edit.css';
 const TOTAL_PAGE_MARGIN = 50;
 const PAGE_BORDER = 1;
 
-class EditorCarousel extends Component {
-	translateWrapper = () => {
+const EditorCarousel = () => {
+	const { pages, currentPage, currentIndex, previousPage, nextPage, isReordering } = useSelect( ( select ) => {
+		const {
+			getBlockOrder,
+			getBlocksByClientId,
+			getAdjacentBlockClientId,
+		} = select( 'core/block-editor' );
+		const { getCurrentPage, isReordering: _isReordering } = select( 'amp/story' );
+
+		const _currentPage = getCurrentPage();
+		const _pages = getBlocksByClientId( getBlockOrder() );
+
+		const index = pages.findIndex( ( { clientId } ) => clientId === currentPage );
```

Comment:
Surely this should be `_pages`, right?

```suggestion
		const index = _pages.findIndex( ( { clientId } ) => clientId === currentPage );
```